packages/transaction-controller/src/TransactionController.ts

```diff
@@ -285,43 +319,238 @@ const controllerName = 'TransactionController';
 /**
  * The external actions available to the {@link TransactionController}.
  */
-type AllowedActions =
+export type AllowedActions =
   | AddApprovalRequest
   | NetworkControllerFindNetworkClientIdByChainIdAction
   | NetworkControllerGetNetworkClientByIdAction;
 
-type AllowedEvents = NetworkControllerStateChangeEvent;
+/**
+ * The external events available to the {@link TransactionController}.
+ */
+export type AllowedEvents = NetworkControllerStateChangeEvent;
+
+/**
+ * Represents the `TransactionController:stateChange` event.
+ */
+export type TransactionControllerStateChangeEvent = ControllerStateChangeEvent<
+  typeof controllerName,
+  TransactionControllerState
+>;
+
+/**
+ * Represents the `TransactionController:incomingTransactionBlockReceived` event.
+ */
+export type TransactionControllerIncomingTransactionBlockEvent = {
+  type: `${typeof controllerName}:incomingTransactionBlock`;
```

Comment:
```suggestion
  type: `${typeof controllerName}:incomingTransactionBlockReceived`;
```

Or the comment above needs to be updated to match